controllers/automation/policyautomation_controller.go

```diff
@@ -381,6 +382,16 @@ func (r *PolicyAutomationReconciler) Reconcile(
 			if len(targetList) > 0 {
 				log.Info("Creating an Ansible job", "targetList", targetList)
 
+				lastAjGeneration, err := common.GetLastAjGeneration(policyAutomation, r.DynamicClient)
```

Comment:
To write down what I recommended earlier. Since the goal is to see if there is an AnsibleJob for this PolicyAutomation of this generation, you don't need to know the latest one, just that there is one that matches the criteria.